### PR TITLE
Require float-save-restore in esp-wifi

### PR DESCRIPTION
**Pull request by bugadani**
For real this time...

### CODE REVIEW

File: esp-wifi/src/preempt/mod.rs [hunk 1]
```diff
@@ -22,7 +22,7 @@ pub fn allocate_main_task() -> *mut Context {
         }
 
         let ptr = malloc(size_of::<Context>() as u32) as *mut Context;
-        core::ptr::write_bytes(ptr, 0, 1);
+        core::ptr::write(ptr, Context::new());
```
Comment (bugadani) [inline esp-wifi/src/preempt/mod.rs:25]: This really should be `Box::new` but I'm not ready to go down that rabbit hole yet.

Comment (bjoernQ) [reply esp-wifi/src/preempt/mod.rs:25]: Ideally we would do construct-in-place

**Comment by bjoernQ** [comment]
This makes `esp-wifi` work on ESP32 for me again but I'm a bit concerned.

Tasks are created on `main` so we initialize cp-enable to enable floats but (as I learnt yesterday) the driver (for ESP32 at least) is using floats in the `ppTask` - so if there is a context switch between two tasks which are both using floats we won't save/restore the float-regs which might cause "interesting" effects?

Maybe the only sane and safe way to use `esp-wifi` (on ESP32 at least) is to require `"float-save-restore"`

**Review by bjoernQ — APPROVED** [review]
Thanks!

**Review by jessebraham — APPROVED** [review]

**Pull request merged**